src/solver/simulation/solver.hxx

```diff
@@ -1609,6 +1613,34 @@ void ISimulation<Impl>::loopThroughYears(uint firstYear,
     }
 }
 
+template<class Impl>
+void ISimulation<Impl>::allocateValeursGenereesParPays()
```

Comment:
I would propose that this method returns the objet instead of initializing it in place, so that it's clear that it's created at this point.